cadence/contracts/bridge/FlowEVMBridgeUtils.cdc

```diff
@@ -268,10 +275,10 @@ contract FlowEVMBridgeUtils {
     /// @return True if the type is either an NFT or FT, false otherwise
     ///
     access(all)
-    view fun isValidFlowAsset(type: Type): Bool {
-        let isFlowNFT = type.isSubtype(of: Type<@{NonFungibleToken.NFT}>())
-        let isFlowToken = type.isSubtype(of: Type<@{FungibleToken.Vault}>())
-        return (isFlowNFT && !isFlowToken) || (!isFlowNFT && isFlowToken)
+    view fun isValidCadenceAsset(type: Type): Bool {
+        let isCadenceNFT = type.isSubtype(of: Type<@{NonFungibleToken.NFT}>())
+        let isCadenceFungibleToken = type.isSubtype(of: Type<@{FungibleToken.Vault}>())
+        return isCadenceNFT != isCadenceFungibleToken
```

Comment:
Technically it is possible for assets to be both fungible and non-fungible tokens, right? Are we not supporting that for simplicity?

Reply:
> Technically it is possible for assets to be both fungible and non-fungible tokens, right?

Not without function overloading in Cadence since both have functions named `deposit` and `withdraw. 

> Are we not supporting that for simplicity?

That's right - mutually exclusive implementations are the only use cases supported for v1. That may change in the future. While it's not currently possible to have a mixed FT/NFT due to the need for function overloading, I added the check here anyway in case it does become technically possible at some point in the future.